autoscheduler/scraper/tests/api_tests.py

```diff
@@ -265,3 +267,48 @@ def test_api_course_search_gives_correct_results_c(self):
         # Assert
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_3(self):
+        """ Tests that /api/course/search?search=CSCE%203&term=201731 gives correct
+            output
+        """
+        # Arrange
+        expected = {'results': ['CSCE 310', 'CSCE 315']}
+        data = {'search': 'CSCE%203', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_lower(self):
+        """ Tests that /api/course/search?search=csce&term=201731 gives correct
+            output
+        """
+        # Arrange
+        expected = {'results': ['CSCE 181', 'CSCE 310', 'CSCE 315']}
+        data = {'search': 'csce', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_3_lower(self):
+        """ Tests that /api/course/search?search=csce%203&term=201731 gives correct
+            output (lowercase search with a number works)
```

Comment:
```suggestion
        """ Tests that /api/course works with lowercase subject and a number works
```

Doesn't have to be this exact phrasing, but there's no reason to say "correct output" when you could instead just say the situation it's actually testing.

Likewise, I honestly don't think the `/api/course/search?search=csce...` stuff is that helpful, and I think it would be more useful if you just dropped the URL parameter stuff and just straight up either said what the situation was like this, or just mention what the parameters are.